echopype/echodata/combine.py

```diff
@@ -154,12 +205,250 @@ def check_echodatas_input(echodatas: List[EchoData]) -> Tuple[str, List[str]]:
+    """
+
+    if channel_selection is None:
+
+        # sort each element in list, so correct comparison can be made
+        all_chan_list = list(map(sorted, all_chan_list))
+
+        # determine if the channels are the same across all Datasets
+        all_chans_equal = [all_chan_list[0]] * len(all_chan_list) == all_chan_list
+
+        if not all_chans_equal:
+
+            # obtain all unique channel names
+            unique_channels = set(itertools.chain.from_iterable(all_chan_list))
+
+            # raise an error if we have varying channel lengths
+            raise RuntimeError(
+                f"For the EchoData group {ed_group} the channels: {unique_channels}, are "
+                f"not provided in each EchoData object being combined. One can select which "
+                f"channels should be included using the keyword argument channel_selection in "
+                f"combine_echodata."
+            )
+
+    else:
+
+        # make channel_selection a set, so it is easier to use
+        channel_selection = set(channel_selection)
+
+        # TODO: if we will allow for expansion, then the below code should be
+        #  replaced with a code section that makes sure the selected channels
+        #  appear at least once in one of the other Datasets
+
+        # determine if channel selection is in each element of all_chan_list
+        eds_num_chan = [
+            channel_selection.intersection(set(ed_chans)) == channel_selection
+            for ed_chans in all_chan_list
+        ]
+
+        if not all(eds_num_chan):
+            # raise a not implemented error if expansion (i.e. padding is necessary)
+            raise NotImplementedError(
+                f"For the EchoData group {ed_group}, some EchoData objects do "
+                f"not contain the selected channels. This type of combine is "
+                f"not currently implemented."
+            )
+
+
+def _create_channel_selection_dict(
+    sonar_model: str,
+    has_chan_dim: Dict[str, bool],
+    user_channel_selection: Optional[Union[List, Dict[str, list]]] = None,
+) -> Dict[str, Optional[list]]:
+    """
+    Constructs the dictionary ``channel_selection_dict``, which specifies
+    the ``channel`` dimension names that should be selected for each
+    ``EchoData`` group. If a group does not have a ``channel`` dimension
+    the dictionary value will be set to ``None``
+
+    Parameters
+    ----------
+    sonar_model: str
+        The name of the sonar model corresponding to ``has_chan_dim``
+    has_chan_dim: dict
+        A dictionary created using an ``EchoData`` object whose keys are
+        the groups of the ``EchoData`` object and values specify if that
+        particular group has a ``channel`` dimension
+    user_channel_selection: list or dict, optional
+        A user provided input that will be used to construct the values of
+        ``channel_selection_dict`` (see below for further details)
+
+    Returns
+    -------
+    dict
```

Comment:
```suggestion
    channel_selection_dict : dict
```

Reply:
I added this variable name here because it is referred to under one of the Parameters.

Also, what do you think to add a line like "See tests/echodata/test_echodata_combine.py::test_create_channel_selection_dict" for example outputs from this method." maybe as a Note?

Reply:
> I added this variable name here because it is referred to under one of the Parameters.

Indeed, this variable name should be included, thank you for catching that!

> Also, what do you think to add a line like "See tests/echodata/test_echodata_combine.py::test_create_channel_selection_dict" for example outputs from this method." maybe as a Note?

That sounds like a good note that would be helpful in the future. I will go ahead and add it in the `Notes` section of the docstring as you suggested.